simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/StationComponentsHolder.java

```diff
@@ -39,4 +41,16 @@ public StationComponentsHolder(Station station) {
     public Optional<StationComponent> getComponent(String componentName) {
         return Optional.ofNullable(components.get(componentName));
     }
+
+    /**
+     * Generates report data for all components in the holder
+     *
+     * @param onlyMutableVariables if true, returns only those variables that can be set by the operator
+     * @return list of {@link ReportDatum}
+     */
+    public List<ReportDatum> generateReportData(boolean onlyMutableVariables) {
+        List<ReportDatum> reportData = new ArrayList<>();
+        components.values().forEach(component -> reportData.addAll(component.generateReportData(onlyMutableVariables)));
+        return reportData;
```

Comment:
Can be simplified: `components.values().stream().flatMap(component -> ...).collect(toList())`